courses/utils.py

```diff
@@ -158,6 +154,31 @@ def generate_program_certificate(user, program):
     return program_cert, True
 
 
+def get_or_create_program_enrollment(user, program):
+    """
+    Get or create new program enrollment.
+
+    Args:
+        user (User): a Django user.
+        program (programs.models.Program): program where the user is enrolled.
+    Returns:
+        (ProgramEnrollment, bool): A tuple containing a
+        ProgramEnrollment object paired with a boolean
+        indicating whether the enrollment was newly created.
+    """
+    if not ProgramEnrollment.objects.filter(program=program, user=user).exists():
+        return (
+            ProgramEnrollment.objects.create(
+                program=program, user=user, active=True, change_status=None
+            ),
+            True,
+        )
+    return (
+        ProgramEnrollment.objects.filter(program=program, user=user).first(),
+        False,
+    )
+
```

Comment:
> Nitpick
> 
> ```
> existing_program_enrollment = ProgramEnrollment.objects.filter(program=program, user=user).first()
> if existing_program_enrollment is None:
>     ------
> 
> return (
>      existing_program_enrollment,
>      False,
> )
> ```

I think `if existing_program_enrollment is None:` is unnecessary as I'm already checking this on the above condition 
``` 
if not ProgramEnrollment.objects.filter(program=program, user=user).exists():
    return (
        ProgramEnrollment.objects.create(
            program=program, user=user, active=True, change_status=None
        ),
        True,
    )
```
@rachellougee Please let me know your thoughts, thanks

Reply:
I wasn't too clear, I think I meant that `ProgramEnrollment.objects.filter(program=program, user=user)` is used in two separate lines- line 169 and line 177, so I suggested making `ProgramEnrollment.objects.filter(program=program, user=user)` to a variable so we can use it in the if the condition on line 169 and return value on line 177. But since it's going to be triggered one way or the other, so it doesn't matter that much.

Reply:
@rachellougee I've made some changes please review it again.